hq/test/aws/ec2/EC2Test.scala

```diff
@@ -140,6 +142,38 @@ class EC2Test extends FreeSpec with Matchers with Checkers with PropertyChecks
     }
   }
 
+  "VPC" - {
+    val sgs1 = SGOpenPortsDetail("Ok", "eu-west-1", "name-1", "id-122", "vpc-1", "tcp", "1099", "Yellow", false )
+    val sgs2 = SGOpenPortsDetail("Ok", "eu-west-2", "name-2", "id-122", "vpc-2", "tcp", "1099", "Yellow", false )
+    val sgs3 = SGOpenPortsDetail("Ok", "eu-west-2", "name-3", "id-122", "vpc-3", "tcp", "1099", "Yellow", false )
+
+    val vpcsMap = Map(
+      "vpc-1" -> new Vpc().withVpcId("vpc-1").withTags(new Tag("Name", "name-1")),
+      "vpc-2" -> new Vpc().withVpcId("vpc-2").withTags(new Tag("Name", "name-2")),
+      "vpc-3" -> new Vpc().withVpcId("vpc-3")
+    )
+    val sgsList = List(sgs1, sgs2, sgs3)
+
+    "get vpcs"  in {
```

Comment:
When you're writing `FreeSpec` tests the test message is the concatenation of all the blocks that lead to the test. In this case that is `"VPC"` and `"get vpcs"`, resulting in a test message of `"VPC get vpcs"`. This is not a helpful test message!

I generally create blocks to contain tests grouped by functionality. This means the test setup can be shared as well as giving the file logical grouping based on the scope. Within each method it can be helpful to group tests there as well for the same reasons.

None of that is set in stone, `FreeSpec` is deliberately flexible! However, whatever test runner you use you do need to make sure the test describes what you expect to happen, so that when it fails the developer has some context about what they'll need to fix.

An example from elsewhere in this file is `parseDescribeNetworkInterfacesResults returns nothing if no interfaces match search IDs` which describes what we're after (and shares some test setup in the scope as I've mentioned above).